pipeline/pipeline.go

```diff
@@ -42,6 +44,13 @@ type Context struct {
 	Reconciler     reconcile.Reconciler
 }
 
+func (c Context) Context() context.Context {
+	if c.Ctx != nil {
+		return context.Background()
```

Comment:
I'd prefer to remove this bandaid. Contexts should always be explicitly defined in my mind and not as a side effect.

I guess it is only used in tests?

Reply:
Good point. I was unsure whether I should add this. It's just with the way we use the `pipeline.Context` it seems very easy to forget setting the actual context. But I guess we have the same problem anyway if we forget any of the other fields..